Add three plots with stellar birth temperatures

EvgeniiChaikin opened the pull request:
Example here: https://swift.dur.ac.uk/COLIBREPlots/chaikin/comparisons/YEAR2023/z0.0_L25N376_weak_SN_and_AGN_feedback/index.html

![Screenshot from 2024-01-12 16-03-05](https://github.com/SWIFTSIM/pipeline-configs/assets/20153933/40548bcf-b2aa-4a3d-8a9d-9932dc02cfec)

EvgeniiChaikin commented:
Note that this PR follows https://github.com/SWIFTSIM/velociraptor-python/pull/106

robjmcgibbon commented on colibre/registration.py:
```diff
@@ -1690,6 +1690,42 @@ def register_stellar_birth_densities(self, catalogue):
+
+    try:
+        average_log_T = catalogue.get_quantity("stellar_birth_temperatures.logaverage")
+        stellar_mass = catalogue.get_quantity("apertures.mass_star_100_kpc")
```
Should this be 50kpc instead of 100, since that's what we're plotting? register_stellar_birth_densities would also need updated

EvgeniiChaikin replied:
Indeed! I've fixed it.

robjmcgibbon commented on colibre/registration.py:
```diff
@@ -1690,6 +1690,42 @@ def register_stellar_birth_densities(self, catalogue):
     return
 
 
+def register_stellar_birth_temperatures(self, catalogue):
```
Could we combine this function with register_stellar_birth_densities?

EvgeniiChaikin replied:
Done

robjmcgibbon commented on colibre/auto_plotter/stellar_birth_properties.yml:
```diff
@@ -115,3 +115,90 @@ stellar_birth_density_stellar_mass_max_all_50:
+      units: Solar_Mass
+  metadata:
+    title: Average of log Stellar Birth Temperature-Stellar Mass relation
+    caption: Includes all haloes, including subhaloes.
```
Joop has requested we add some more info to the captions

EvgeniiChaikin replied:
Done

EvgeniiChaikin replied:
Done

robjmcgibbon commented:
I just pushed a minor change to address [this comment by Joop](https://leiden-observatory.slack.com/archives/G035RUR984V/p1705313259121299)

EvgeniiChaikin commented:
@robjmcgibbon  I have added plots showing stellar birth pressures. This update follows https://github.com/SWIFTSIM/SOAP/pull/62 and https://github.com/SWIFTSIM/velociraptor-python/pull/107

Now for all stellar birth properties -- density, temperature, and pressure -- we plot

- the minimum value
- the maximum value
- the median value

These are plotted vs. stellar mass. I removed the plots vs. halo mass because otherwise we would have way too many plots.

robjmcgibbon reviewed (commented):
Looks good to me, but could you please post a link with the new plots? I'm curious to see how the median compares with log-of-average

robjmcgibbon reviewed (approved):
Cool thanks, looks like it's pretty similar to log-of-average
